src/Middleware/LoginSessionMiddleware.php

```diff
@@ -35,8 +37,11 @@ public function process(HTTPRequest $request, callable $delegate)
             // If the session has already been revoked, or we've got a mismatched
             // member / session, log the user out (this also revokes the session)
             if (!$loginSession || (int)$loginSession->MemberID !== (int)$member->ID) {
-                $identityStore = Injector::inst()->get(IdentityStore::class);
-                $identityStore->logOut($request);
+                Config::withConfig(function ($config) use ($request) {
+                    $config->set(RememberLoginHash::class, 'logout_across_devices', false);
```

Comment:
This seems bad. You basically rely on knowledge of the inner implementation of `RememberLoginHash` to decide to temporarily hack the config.

I would rather just set `logout_across_devices` to `false` all the time. If that's not advisable, then we should update `RememberLoginHash` so we can prevent this behaviour without temporarily changing a config flag.

Reply:
I'm thinking we could just manually destroy the session here and unset the current user. Your `RememberLogin` should already have been deleted by the terminate session call on the other machine.

We could also attach some bit of data to the `HTTPRequest`. Logout handlers could conditionally use that bit of data to do things.

Reply:
Is there a scenario where we want to support `logout_across_devices` to be `true`?

Reply:
> Is there a scenario where we want to support logout_across_devices to be true?

I think so, to retain the existing behavior where if you manually log out of a device to regular way, you log out all devices.

If we choose still change the default value to false which will also affect people without session manager, then that's fine.  The way this PR is structured it now works independently of whether or not we have `logout_across_devices` set to true or false in framework.

> I'm thinking we could just manually destroy the session here and unset the current user ...

I'm not sure about we should do this?  It seems like we'd be changing `$identityStore->logOut($request);` to something else?  `$identityStore->logOut($request);` seems like it works correctly right now - I don't understand why we would want to change this?

Also the conditional starts with `if (!$loginSession ||` so, the $loginSession has already been destroyed?

> We should update RememberLoginHash so we can prevent this behaviour without temporarily changing a config flag

Agree that updating the public API on RememberLoginHash seem a bit more correct then hacking config to update internals in RememberLoginHash.  I'd still like to keep the core approach taken here where we temporarily disable the logout_across_devices.  I like the idea of taking an approach that has the lightest touch and least potential side-effects.

> we should update RememberLoginHash so we can prevent this behaviour without temporarily changing a config flag.

Would this approach work?

**RememberLoginHash.php**
```php

/** @var bool */
protected static $logoutAcrossDevices;

public static function getLogoutAcrossDevices(): bool
{
    if (is_bool($this->logoutAcrossDevices)) {
        return $this->logoutAcrossDevices;
    }
    return static::config()->get('logout_across_devices');
}

public static function setLogoutAcrossDevices(bool $value): void
{
    $this->logoutAcrossDevices = $value;
}

public static function clear(Member $member, $alcDevice = null)
{
    ...
-       if (static::config()->logout_across_devices) {
+      if (static::getLogoutAcrossDevices()) {
            self::get()->filter(['MemberID' => $member->ID])->removeAll();
    ...
}    
```

**LoginSessionMiddleware.php**
```php
$default = RememberLoginHash::getLogoutAcrossDevices();
RememberLoginHash::setLogoutAcrossDevices(false);
$identityStore = Injector::inst()->get(IdentityStore::class);
$identityStore->logOut($request);
RememberLoginHash::setLogoutAcrossDevices($default);
```

?

Reply:
The conclusion from https://github.com/silverstripe/silverstripe-framework/issues/9794 was that `logout_across_devices` should default to `true` if you don't have session manager installed because you don't have a way an obvious way of invalidating RememberLoginHashs on other devices. However, when you have session manager installed, you have a good way of invalidating remember me tokens, so we should default `logout_across_devices` to `false` because it provides a better user experience.

I do agree that when the user uses session manager to terminate a specific session, `logout_across_devices` shouldn't come into play because the user is not technically logging out from the current session.

Having a method like `RememberLoginHash::setLogoutAcrossDevices(false)` is probably a good compromise. I'm thinking we probably don't need to reset the value back to its default after we've logout, because there won't be any CurrentUser for the rest of the request any way. So we could call the method something like `RememberLoginHash::supressTokenClearing()` and not require any parameter.

An alternative could be to have `LoginSessionMiddleware` add a magic header to the HTTPRequest (e.g.: `$request->addHeader('force-logout', 1)`) and update `CookieAuthenticationHandler::logOut()` to only call `RememberLoginHash::clear()` if `force-logout` is falsy. But, I wouldn't say this is an obviously better solution than what you are suggesting.

Either way, I think it's better to add a tiny bit of code to framework to make this behaviour work better.

Reply:
Re changing the default value of `logout_across_devices`, we can still do this if we want, though we do need to consider the use case where a developer just changes the default i.e. if we make it false if session manager is installed, the developer overrides this on their project to true, session-manager should still function correctly.

`RememberLoginHash::supressTokenClearing()` is cleaner for this use case, though my concern here would be for future unforeseen use-cases, the classic get/set functions do provide more flexibility

I'll do a PR for framework like the code sample provided above and update this PR.  Once that's up we can see how we feel

Reply:
@maxime-rainville have created framework PR https://github.com/silverstripe/silverstripe-framework/pull/9917

I've updated this PR to use it, though the framework PR code won't be available in a tagged release until 4.8.0.  Session manager needs to be able to work with framework 4.x-dev and also 4.7.x-dev in the meantime (minimum constraint in composer.json in framework ^4.7). I don't think there's a composer constraint available that covers these 3 scenarios.

So I've had to resort to using a `method_exists()` and a `TODO` to remove it, which is pretty ugly unfortunately.  If we go with this we should add a post quarterly release task to remove this code.